app/controllers/application_pets_controller.rb

```diff
@@ -0,0 +1,27 @@
+class ApplicationPetsController < ApplicationController
+  def update
+    application = Application.find(params[:id])
+    selected_app_pet = ApplicationPet.find(params[:pet_id])
+    if params[:approved?] == "Yes"
+      selected_app_pet.update(application_pet_status: 2)
+      selected_app_pet.update(adoptable: false)
+    else
+      selected_app_pet.update(application_pet_status: 3)
+      selected_app_pet.pet.update(adoptable: true)
+    end
+    update_application(application)
+    redirect_to "/admin/applications/#{application.id}"
+  end
+
+  def update_application(application)
```

Comment:
Nice delegation to a helper method! I think this could potentially live in the Application model to allow for more easily unit testing it